flocker/control/schema/types.yml

```diff
@@ -59,7 +59,17 @@ definitions:
         external:
           type: integer
     uniqueItems: true
-    
+
+  environment:
+    title: "Container environment variables"
+    description: "Mapping of label and value pairs for environment variables"
+    type: object
+    patternProperties:
+      "^.{1,4096}$":
+        type: string
+        maxLength: 4096
+    additionalProperties: false
```

Comment:
@exarkun save me some Googling :smile:  how does one limit the number of arbitrary properties in a JSON schema object? And what would you regard as a sensible limit?